crates/bevy_image/src/dynamic_texture_atlas_builder.rs

```diff
@@ -67,22 +81,30 @@ impl DynamicTextureAtlasBuilder {
         atlas_texture: &mut Image,
         allocation: Allocation,
         texture: &Image,
-    ) {
+    ) -> Result<(), DynamicTextureAtlasBuilderError> {
         let mut rect = allocation.rectangle;
         rect.max.x -= self.padding as i32;
         rect.max.y -= self.padding as i32;
         let atlas_width = atlas_texture.width() as usize;
         let rect_width = rect.width() as usize;
         let format_size = atlas_texture.texture_descriptor.format.pixel_size();
 
+        let Some(ref mut atlas_data) = atlas_texture.data else {
+            error!("Atlas texture has no texture data");
+            return Err(DynamicTextureAtlasBuilderError::UninitializedAtlas);
+        };
+        let Some(ref data) = texture.data else {
+            error!("Source texture provided has no texture data");
```

Comment:
I think we should just let the error bubble up to the user and let them decide if they want to log it.